packages/@uppy/core/types/index.d.ts

```diff
@@ -126,6 +126,7 @@ export type PluginTarget =
   | Element
   | typeof BasePlugin
   | typeof UIPlugin
+  | BasePlugin
```

Comment:
We have `typeof BasePlugin` but also `BasePlugin`? Should be perhaps be `InstanceOf<typeof BasePlugin>`?

Reply:
I'm not familiar with the nuances of that, I'd need to look it up. But basically what we were missing was that the user can either pass the plugin class, or a plugin instance. Can `InstanceOf<>` target both at once?

Reply:
Theoretically I think it would satisfy both, but technically it means the initialized instance. Regardless, if we want to type the instance, there should at least be a `InstanceOf` type.

Reply:
`Cannot find name 'InstanceOf'`, also it doesn't seem to exist, or what am I missing?

Reply:
Can we try to replace both with this? I think it should be stricter than current attempt

https://www.typescriptlang.org/docs/handbook/utility-types.html#instancetypetype

Reply:
The docs says that `InstanceType<typeof C>` is `C` though, what's stricter about it? I just tried in the playground, and it seems `InstanceType` doesn't  help us here: https://www.typescriptlang.org/play?#code/MYGwhgzhAEDC0G8BQ1oA9oF5oAYDcK0AnlrgQL4pIAuRADgKbQAqOpAkgHYTVifANm9BgB5ajAPYAzOAD4C4pswCMpWAuEsATGugAfaIulwCSZMAndqhhjxwAuFm2ycGAdzhILVmz2WPWNSRKM0JUbx5fagcWVRd3Twjraltqf1igkORUcMtIlLsAnVJXD1gvPOTU9OYdbHKQpCA

Reply:
Thanks for checking. I remembered this helped me with typing tus-node-server, perhaps I remember wrong. This is okay then!

Reply:
That being said, `typeof` doesn't seem to behave as we are trying to make it behave: https://www.typescriptlang.org/play?#code/MYGwhgzhAEDC0G8BQ1oA9oF5oAYDcK0AnlrgQL5JKiQwAi0ApmgC6MB2AJjPApdeCjQAookKoM2AIwFUqEtgDMFFEhZEADo2gAVHKXVaA9gDM40AD7RDjU9DprN2nVINO7sAjd0AmN8bM6AiRkYCN2CBZrRkicAC5dfWxYanDI6MipBL1SBzCIqLZIn2ykkSR+ZDlofPSilnjdV2TUgoyWLKbc1rqYlhLE0mEKqiq5WsK+xp0-bon2zpnSFPn6gZzsYf4kIA